ShoppingM8-be/src/main/java/edu/pw/shoppingm8/product/api/ProductControler.java

```diff
@@ -0,0 +1,113 @@
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "If valid data was provided"),
+        @ApiResponse(code = 400, message = "If invalid data was provided"),
+        @ApiResponse(code = 403, message = "If user does not have access to the list"),
+        @ApiResponse(code = 404, message = "If invalid list id was provided")})
+    @PostMapping
+    public ResponseEntity<Void> createProduct(
+        @PathVariable Long listId, @Valid @RequestBody ProductCreateRequestDto productDto) {
+        Product product = productService.createProduct(listId, productDto);
+        return ResponseEntity.created(
+            URI.create(String.format("/list/%d/product/%d", listId, product.getId())))
+            .build();
+    }
+    
+    @ApiOperation(value = "Patch product", nickname = "patch product", notes = "",
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "If valid data was provided"),
+        @ApiResponse(code = 400, message = "If invalid data was provided"),
+        @ApiResponse(code = 403, message = "If user does not have access to the list"),
+        @ApiResponse(code = 404, message = "If invalid id was provided")})
+    @PatchMapping("{id}")
+    public ResponseEntity<Void> patchProduct(
+        @PathVariable Long listId, @PathVariable Long id, @Valid @RequestBody ProductPatchRequestDto productDto) {
+        productService.patchProduct(listId, id, productDto);
+        return ResponseEntity.ok().build();
+    }
+    
+    @ApiOperation(value = "Purchase product", nickname = "puchase product", notes = "",
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "If valid data was provided"),
+        @ApiResponse(code = 403, message = "If user does not have access to the list"),
+        @ApiResponse(code = 404, message = "If invalid id was provided")})
+    @PostMapping("{id}")
+    public ResponseEntity<Void> purchaseProduct(@PathVariable Long listId, @PathVariable Long id, @RequestParam Double amount) {
+        productService.purchaseProduct(listId, id, amount);
+        return ResponseEntity.ok().build();
+    }
+    
+    @ApiOperation(value = "Delete product", nickname = "delete product", notes = "",
+            authorizations = {@Authorization(value = "JWT")})
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "If valid id was provided"),
+        @ApiResponse(code = 403, message = "If user does not have access to the list"),
+        @ApiResponse(code = 404, message = "If invalid id was provided")})
+    @DeleteMapping("{id}")
+    public ResponseEntity<Void> deleteProduct(@PathVariable Long listId, @PathVariable Long id) {
+        productService.deleteProduct(listId, id);
+        return ResponseEntity.ok().build();
```

Comment:
How about NO_CONTENT status? Since the requested resource is no longer accesible and no body is returned.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE

Reply:
fixed